### ISSUE TITLE
4K UHD 60p playback issue in Chrome and Firefox

### ISSUE THREAD

**Issue by Kjeld** [videojs/videojs-contrib-hls#1309]
## Description
The issue is that our 4K UHD stream doesn't play video. It only reads (and plays back) the audio .ts chucks. We think it is related to a undefined width and height issue.

## Sources
https://content.streamone.io/embed/item=TpZBf-QKi4e2/bbbuhd60p.html
https://media.streamone.io/hls/account=MpBJPKUJ4Y6i/item=TpZBf-QKi4e2/TpZBf-QKi4e2.m3u8

## Steps to reproduce
1.  In Chrome (Version 63.0.3239.84 High Sierra) https://content.streamone.io/embed/item=TpZBf-QKi4e2/bbbuhd60p.html
2. https://videojs.github.io/videojs-contrib-hls/ source https://media.streamone.io/hls/account=MpBJPKUJ4Y6i/item=TpZBf-QKi4e2/TpZBf-QKi4e2.m3u8
3. Check the Chrome developer Tools

### Expected
Playback of 60fps UHD video.

### videojs-contrib-hls version
It can be reproduced on you latest release
https://videojs.github.io/videojs-contrib-hls/

### Browsers
Chrome
Firefox

### Platforms
High Sierra
Debian 8

PS: The item is currently transcoding on a test account for you to publicly access, it will be there in 60 minutes from now. 12:23 UTC+1 == NOW()

**Comment by forbesjo** [comment]
This is related to https://github.com/videojs/videojs-contrib-hls/issues/1435, closing this in favor of the other issue

**Comment by squarebracket** [comment]
Reopening as it's unrelated to #1435 (segment size isn't what's causing the describing problem per se).

@Kjeld your playlist confuses me. You have an `EXT-X-MEDIA` for the audio with no `URI`, but then have an audio-only `EXT-X-STREAM-INF` later. Is there a reason you're not just using it as the `URI` attribute of the `EXT-X-MEDIA`?

In any case, what's happening here is that the initial bandwidth is low enough and the video playlist's `BANDWIDTH` is high enough that the initial playlist selector is choosing the audio-only playlist by default. Once that happens, it's locked into audio-only; the source buffers are made only once, and since the first playlist it selects only has audio, it only creates an audio source buffer.

If you feed in a pre-programmed `bandwidth` to use during initial playlist selection, à-la

```javascript
videojs('videojs-contrib-hls-player', {html5: {hls: {bandwidth: 30000000}}})
```

it will select the audio+video playlist and everything will work out.

Perhaps we should add something to the initial playlist selection that disregards any audio-only variants if there are any audio+video variants.

That being said, unless you need it, you should consider dropping the only audio-only variant playlist. Note that Apple no longer recommends using an audio-only playlist in its HLS authoring guidelines; it now recommends a playlist with low-bitrate audio and video.

**Comment by forbesjo** [comment]
Moving this to VHS and we can fix this initial playlist selection logic there

/move to http-streaming
